docs/developer/decision-records/2024-10-09-vault-authentication-refactor/README.md

```diff
@@ -0,0 +1,244 @@
+As such, `headers` are no longer saved as a variable inside the `HashicorpVaultClient` and `getHeaders()` is called instead to always fetch the newest token.
+
+### HashiCorp Vault Extension refactor
+
+The token based authentication logic is refactored and moved into its own extension named `HashiCorp Vault Token Auth`.
+This includes the token refresh functionality and will lead to a refactoring of the following classes:
+
+`HashicorpVaultExtension.java`: `initialize()` no longer will start the token refresh and `stop()` will be removed since the extension does not govern token refresh anymore.
+
+`HashicorpVaultClient.java`: `isTokenRenewable()` and `renewToken()` and their private methods will be moved to a new dedicated client in the Token Auth Extension.
+
+`HashicorpVaultSettings.java`: some settings are moved and two new ones are added as described in the sections [HashiCorp Vault Token Auth Extension Settings](#hashicorp-vault-token-auth-extension-settings) and [Configuration](#configuration).
+
+`HashicorpVaultTokenRenewTask.java`: is moved to the HashiCorp Vault Token Auth Extension in its entirety.
+
+The respective tests covering the moved functionality are also moved accordingly.
+
+### HashiCorp Vault Token Auth Extension
+
+The HashiCorp Vault Token Auth Extension will contain the token storage functionality and the token refresh functionality that has been removed from the HashiCorp Vault Extension.
+
+#### HashiCorp Vault Token Auth Extension Settings
+
+The following settings have been moved from the HashiCorp Vault Extension to th HashiCorp Vault Token Auth Extension.
+
+Duplicate setting from the HashiCorp Vault Extension, since the HashiCorp Vault Token Auth Extension also needs access to the vault.
+`edc.vault.hashicorp.url`
+
+Settings moved from the HashiCorp Vault Extension, that only concern the token storage and token refresh.
+`edc.vault.hashicorp.token`
+`edc.vault.hashicorp.token.scheduled-renew-enabled`
+`edc.vault.hashicorp.token.ttl`
+`edc.vault.hashicorp.token.renew-buffer`
+
+### HashiCorp Vault Health Extension
+
+The `get()` method of `HashicorpVaultHealthCheck` is currently using a merge of the response from the Hashicorp Vault Health API and the ability to renew the given token.
+After the refactor, the token renewal is no longer a part of the base `HashicorpVaultExtension`.
+As such the `get()` method is adjusted and will only return the response from the Hashicorp Vault Health API.
+
+## Further Considerations
+
+Currently, the authentication method is chosen through configuration and unable to be changed at runtime.
+If the need arises in the future, the authentication method may be chosen through the request headers instead while keeping the configuration setting as default.
+This option was explicitly left out in this decision record, to keep the refactored code as close to the original as possible in terms of functionality and API.
```

Comment:
I wouldn't add these considerations as they are vague and definitely out of scope.